### PR TITLE
scripts: build and run statically compiled tests

### PR DESCRIPTION
**Pull request by heyitsanthony**
/cc @mischief

### CODE REVIEW

**Comment by mischief** [comment]
does `-test.cpu 1,2,4` even work with < 4 cpus?

**Comment by heyitsanthony** [comment]
@mischief yes

**Comment by gyuho** [comment]
LGTM. /cc @xiang90

File: scripts/static-tests [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		echo running $t....
+		./$t -test.v -test.cpu 1,2,4
+	done
+	popd
+	exit 0
+fi
+
+export GO15VENDOREXPERIMENT=1
+
+# use vendoring for building tests
```
Comment (mischief) [inline scripts/static-tests]: only execute this code during build?

File: scripts/static-tests [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+export GOPATH=${PWD}/gopath
+rm -f $GOPATH/src
+mkdir -p $GOPATH
+ln -s ${PWD}/cmd/vendor $GOPATH/src
```
Comment (mischief) [inline scripts/static-tests]: trap rm -rf $GOPATH?

File: scripts/static-tests [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+# Some options that were useful for arm64:
+# GOARCH=arm64 ETCD_UNSUPPORTED_ARCH=arm64 ./scripts/static-tests
+
+set -eo pipefail
+
+OUTDIR="static-tests"
+
+if [ -e "$OUTDIR" ]; then
+	echo "found "$OUTDIR"/ running tests..."
```
Comment (mischief) [inline scripts/static-tests]: when executing the tests, it seems `default.etcd/` is created, so you should trap rm -rf default.etcd here i think.

File: scripts/static-tests (hunk 1 above)
Comment (heyitsanthony) [inline scripts/static-tests]: I think the `exit 0` in the test part covers that?

Comment (mischief) [inline scripts/static-tests]: something has created `gopath/` on the system that is executing the tests.

Comment (heyitsanthony) [inline scripts/static-tests]: OK, I'll take a look at it. Could be leftovers from the old version of script?

**Comment by mischief** [comment]
actually, i apologize, i am still running your old copy of the script (`carepkg.sh`) so i have not actually executed this pr :-(.

still, i think you should trap to cleanup any stuff created by the execution of the build or the tests.

**Comment by heyitsanthony** [comment]
@mischief sure, fair enough

**Comment by xiang90** [comment]
@mischief Can you take a look? Do you still need this for arm testing?

**Comment by pgray** [comment]
Any movement here?

**Pull request closed without merging**